### DISCUSSION TITLE
Submit listings item patch request

### DISCUSSION THREAD

**Discussion by henryvuong2** [amzn/selling-partner-api-models#3769] [General]
I have successfully submitted listings get request, but not sure how to submit patch request. I have read through this article:
https://developer-docs.amazon.com/sp-api/docs/listings-items-api-v2021-08-01-use-case-guide#step-1-submit-listings-item-patch-request
I am trying to change the price of an item. This is my code:

`        public static void Main()

        {
            try
            {
                LWAAuthorizationCredentials lwaAuthorizationCredentials = new LWAAuthorizationCredentials
                {
                    ClientId = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
                    ClientSecret = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
                    RefreshToken = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
                    Endpoint = new Uri("https://api.amazon.com/auth/o2/token")
                };

                ListingsApi listingsApi = new ListingsApi.Builder()
                    .SetLWAAuthorizationCredentials(lwaAuthorizationCredentials)
                    .Build();

                string sellerId = "XXXXXXXXXXX";
                List<string> marketplaceIds = new List<string>();
                marketplaceIds.Add("ATVPDKIKX0DER");
                string sku = "XXX-YYY-ZZZ";
                
		//*** This is the part where I got error
		//*** I am not sure if I am doing correctly
               //*** I want to change the price of the item
		ListingsItemPatchRequest body = new ListingsItemPatchRequest();
                body =  {
                    "productType":"CELLULAR_PHONE",
                      "patches":[

                        {
                        "op":"replace",
                          "path":"/attributes/purchasable_offer",
                          "value":[
                            {
                            "marketplace_id": "ATVPDKIKX0DER",
                              "currency": "USD",
                              "our_price": [
                                {
                                "schedule": [
                                                      {
                                    "value_with_tax": 15.00
                                                      }
                                  ]
                                }
                              ]
                            }
                          ]
                        }

                      ]
                    }

                ListingsItemSubmissionResponse result = listingsApi.PatchListingsItem(sellerId, sku,  marketplaceIds, body);
                Console.WriteLine(result.ToJson());
                Console.ReadLine();
            }
            catch (LWAException e)
            {
                Console.WriteLine("LWA Exception when calling SellersApi#getMarketplaceParticipations");
                Console.WriteLine(e.getErrorCode());
                Console.WriteLine(e.getErrorMessage());
                Console.WriteLine(e.Message);
            }
            catch (ApiException e)
            {
                Console.WriteLine("Exception when calling SellersApi#getMarketplaceParticipations");
                Console.WriteLine(e.Message);
            }

        }`

The "body" parameter is where I got stuck. I am not sure If I do it right. Please advise.

**Comment by rodrifed** [comment]
Hi @henryvuong2,

In the following link you will find an example of a PATCH request: https://github.com/amzn/selling-partner-api-samples/blob/main/use-cases/pricing/code/java/src/main/java/lambda/SubmitPriceHandler.java#L50-L70.
The method updates an existing `purchasable_offer` node, and then creates the corresponding `ListingsItemPatchRequest` object.

I hope the sample code above will help you finalize your code.

Thank you,
Federico
Selling Partner Developer Services

**Reply by henryvuong2** [reply]
I have re-written my code and posted a renewed question here. It is similar to this one but the code is modified. Please take a look: 
https://github.com/amzn/selling-partner-api-docs/discussions/3570
